test/unit/test_parameter_parser.py

```diff
@@ -0,0 +1,512 @@
+import json
+from textwrap import dedent
+from unittest import mock
+
+import pytest
+from pyexasol import ExaConnection
+
+from exasol.ai.mcp.server.parameter_parser import (
+    FuncParameterParser,
+    ScriptParameterParser,
+)
+from exasol.ai.mcp.server.server_settings import MetaParameterSettings
+
+
+@pytest.fixture
+def param_config() -> MetaParameterSettings:
+    return MetaParameterSettings(
+        enable=True,
+        name_field="name",
+        type_field="type",
+        input_field="inputs",
+        emit_field="emits",
+        return_field="returns",
+    )
+
+
+@pytest.fixture
+def func_parameter_parser(param_config) -> FuncParameterParser:
+    return FuncParameterParser(
+        connection=mock.create_autospec(ExaConnection),
+        conf=param_config,
+        tool_name="test_tool",
+    )
+
+
+@pytest.fixture
+def script_parameter_parser(param_config) -> ScriptParameterParser:
+    return ScriptParameterParser(
+        connection=mock.create_autospec(ExaConnection),
+        conf=param_config,
+        tool_name="test_tool",
+    )
+
+
+@pytest.mark.parametrize(
+    ["params", "expected_result"],
+    [
+        (
+            " Pa1 INT , _var_PAR  varchar( 100 ) ",
+            [
+                {"name": "Pa1", "type": "INT"},
+                {"name": "_var_PAR", "type": "varchar( 100 )"},
+            ],
+        ),
+        (
+            '"param1"  DECIMAL(3,2),"PARAM2" decimal(10, 0)',
+            [
+                {"name": "param1", "type": "DECIMAL(3,2)"},
+                {"name": "PARAM2", "type": "decimal(10, 0)"},
+            ],
+        ),
+        (
+            "p1 varchar(3), ts2 timestamp(6 ) with local time zone, d3 decimal(10,5), "
+            "h4 hashtype(4 byte),i5 interval year ( 5) to month",
+            [
+                {"name": "p1", "type": "varchar(3)"},
+                {"name": "ts2", "type": "timestamp(6 ) with local time zone"},
+                {"name": "d3", "type": "decimal(10,5)"},
+                {"name": "h4", "type": "hashtype(4 byte)"},
+                {"name": "i5", "type": "interval year ( 5) to month"},
+            ],
+        ),
+        ('"P_1" INT', [{"name": "P_1", "type": "INT"}]),
```

Comment:
I think,

('"1_P" INT', [{"name": "1_P", "type": "INT"}]),

would be also allowed, but maybe something to try out before